### ISSUE TITLE
CSG function _update_shape() is not documented

### ISSUE THREAD

**Issue by Saitodepaula** [godotengine/godot#39706]
**Your Godot version:**
3.2.1 stable official, on Windows 10

**Issue description:**
[CSGShape _update_shape()](https://github.com/godotengine/godot/blob/master/modules/csg/csg_shape.cpp) is not documented.

When dealing with CSG from code, it looks like this is a very important function (examples of people using and discussing about this function [here](https://github.com/godotengine/godot/issues/29275), [here](https://github.com/godotengine/godot/issues/33138) and [here](https://github.com/godotengine/godot/issues/19232)) . Without it, many things are not possible, so why it is not documented?

**URL to the documentation page (if already existing):**
https://docs.godotengine.org/en/3.2/classes/class_csgshape.html?rtd_search=_update_shape#class-csgshape-method-get-meshes

*Bugsquad edit: Transferred to the main repository as this issue pertains to the class reference.*

**Comment by Xrayez** [comment]
The method is not documented because it's considered to be internal (`_` prefix is used to determine whether the method is going to be exposed). But it's still possible to call these kind of methods, see godotengine/godot-proposals#2285.

So, it's possible to expose the method publicly instead, something like `force_update_shape()` like in ray cast nodes, but we'd have to be sure that it won't unexpectedly break or crash the engine if used this way.

**Comment by Saitodepaula** [comment]
> The method is not documented because it's considered to be internal (`_` prefix is used to determine whether the method is going to be exposed). But it's still possible to call these kind of methods, see [godotengine/godot-proposals#2285](https://github.com/godotengine/godot-proposals/issues/2285).
> 
> So, it's possible to expose the method publicly instead, something like `force_update_shape()` like in ray cast nodes, but we'd have to be sure that it won't unexpectedly break or crash the engine if used this way.

Hum, Ok, I understand, but I would like to have the funcionality available somehow. So, with #46258 , will this be possible?

**Comment by Xrayez** [comment]
Yes, I think #46258 would make it easier to use to get immediate results from CSG operations. But I don't know what you personally need this for, so adding `force_update_shape()` may be enough in fact.

Note that this is just my 2 cents, I haven't used those CSG features myself extensively to fully understand the existing limitations.

**Comment by trollodel** [comment]
@Xrayez You are too fast 😀

> Yes, I think #46258 would make it easier to use to get immediate results from CSG operations.

@Saitodepaula Keep in mind that `_update_shape()` is intended to be used in idle time and I have not removed this limitation.
But if you just need the final mesh, then it will be possible with #46258

**Comment by KoBeWi** [comment]
The methods seems to never be called. It should be removed.
